velocity_modelling/tools/generate_basin_wiki.py

```diff
@@ -0,0 +1,260 @@
+"""
+Generate Markdown files for basins from nzcvm_registry.yaml
+
+This script reads the nzcvm_registry.yaml file and generates a Markdown file for each unique basin name.
+The Markdown file contains details such as the basin type, author, images, notes, boundaries, surfaces, and smoothing boundaries. The generated files are saved in the wiki/basins directory.
+
+Usage:
+    python generate_basin_wiki.py [--registry <path>] [--scale-images]
+
+By default, the script reads the nzcvm_registry.yaml file from the parent directory.
+The --registry argument can be used to specify a different path.
+If the --scale-images flag is provided, images are scaled to 50% size with a clickable link to the full-size image.
+
+Output .md files are saved in the wiki/basins directory.
+
+"""
+
+import argparse
+import re
+from datetime import datetime
+from pathlib import Path
+
+import pytz
+import yaml
+
+# Set up argument parser
+parser = argparse.ArgumentParser(
```

Comment:
Should use typer, I think this goes for most of the tools scripts so won't mention again